src/Tuura/Concept/Circuit/Basic.hs

```diff
@@ -14,6 +15,8 @@ import Data.Monoid
 -- * s is the type of states
 -- * e is the type of events
 
+data Causality e = AndCausality (e, e) | OrCausality ([e], e) deriving (Ord, Eq, Show)
```

Comment:
Why use an extra pair here? I think `data Causality e = AndCausality e e | OrCausality [e] e` should lead to simpler code.